### ISSUE TITLE
Feature Request: update `CpuUsage` in to tablet `RealtimeStats`

### ISSUE THREAD

**Issue by timvaillancourt** [vitessio/vitess#12579]
### Feature Description

This feature request proposes adding system-level CPU usage to tablet `RealtimeStats`

A `CpuUsage` field [currently exists in `RealtimeStats`](https://github.com/vitessio/vitess/blob/main/go/vt/proto/query/query.pb.go#L5037-L5038):
```golang
    // cpu_usage is used for load-based balancing
    CpuUsage float64 `protobuf:"fixed64,5,opt,name=cpu_usage,json=cpuUsage,proto3" json:"cpu_usage,omitempty"`
```
..but this field is never set/read in the code from what I can tell. This RFC aims to close the loop on what was started here.

## Proposal

This request proposes a system "monitor" in tabletserver for the collection of system/server-level health metrics, currently CPU usage only, as a means to gauge tablet capacity. This collector could be an opt-in `vttablet` feature with a command-line flag

This system health collector would collect averaged, system-level CPU usage. This collector is generic in case further system-level metrics are needed although I don't really expect additions

As gathering CPU usage depends on a blocking sampling period, the cpu usage metrics should be gathered asynchronously and the "getter" will return the most-recent CPU usage result

### Use Case(s)

There are two upcoming RFCs that depend on this metric existing in `RealtimeStats`

I would prefer to rollout the `CpuUsage` metric separately, in advance of those RFCs, but it could also be combined with the 1st change at the cost of a larger PR

### Replica Load-Balancing
```
// cpu_usage is used for load-based balancing
```
As this comment alludes to, the addition of CPU usage to replica load balancing could help distribute load more evenly

The idea is to chose a replica with the least CPU usage when two or more replicas are otherwise considered equal. To prevent a thundering herd to a replica with less CPU load _(plus the metric is only updated periodically)_, this optimization should probably be applied to a portion of traffic only eg: 50-66% :thinking:

### Query Backpressure

Slack is working on tablet-level backpressure/overload protections (see `#feat-handling-overload` channel) using replication lag and CPU usage as a signal of saturation for write transactions and read queries respectively

At a high level: when a configurable CPU threshold is reached, a tablet will refuse certain read workloads (by priority) to shed load that is best-used for higher priority workloads

**Comment by harshit-gangal** [comment]
50-60% utilization is considered a good utilization, upto 70% the CPU works without any issues. 
It is still not clear from the RFC how the algorithm will look like to distribute the load based on CPU as there is an inherent limitation of 1s delay in the updated CPU utilization metric.

**Comment by harshit-gangal** [comment]
I would say we should hold the PR to be merged upstream. But, you can carry out the changes on top of it to see how load distribution will work

**Comment by timvaillancourt** [comment]
> 50-60% utilization is considered a good utilization, upto 70% the CPU works without any issues.

@harshit-gangal agreed, 70% CPU is no concern. When I mentioned 50-60% I did not mean CPU utilisation however, the high-utilisation percent would be configurable in this solution

What I meant with "50-60%" is thundering herd prevention, but maybe it's a pre-optimisation. Consider a scenario where:
- A replica with low CPU utilisation is added to the topology, all other replicas are busy
- A thundering herd chose this new replica because it has much less CPU usage
I wonder if this would create an inbalance in traffic 🤔
